lib/mpl_toolkits/mplot3d/axes3d.py

```diff
@@ -2744,6 +2745,136 @@ def calc_arrow(uvw, angle=15):
 
     quiver3D = quiver
 
+    def voxels(self, filled, color=None, **kwargs):
+        """
+        Plot a set of filled voxels
+
+        All voxels are plotted as 1x1x1 cubes on the axis, with filled[0,0,0]
+        placed with its lower corner at the origin. Occluded faces are not
+        plotted.
+
+        .. versionadded:: 2.1
+
+        Parameters
+        ----------
+        filled : np.array
+            A 3d array of values, with truthy values indicating which voxels
+            to fill
+
+        color : array_like
+            Either a single value or an array the same shape as filled,
+            indicating what color to draw the faces of the voxels. If None,
+            plot all voxels in the same color, the next in the color sequence.
+
+        Any additional keyword arguments are passed onto
+        :func:`~mpl_toolkits.mplot3d.art3d.Poly3DCollection`
+
+        Returns
+        -------
+        faces : dict
+            A dictionary indexed by coordinate, where ``faces[i,j,k]`` is a
+            `Poly3DCollection` of the faces drawn for the voxel
+            ``filled[i,j,k]``. If no faces were drawn for a given voxel, either
+            because it was not asked to be drawn, or it is fully occluded, then
+            ``(i,j,k) not in faces``.
+        """
+        # check dimensions
+        if filled.ndim != 3:
+            raise ValueError("Argument filled must be 3-dimensional")
+
+        # handle the color argument
+        if color is None:
+            color = self._get_patches_for_fill.get_next_color()
+        if np.ndim(color) <= 1:
+            color, _ = np.broadcast_arrays(
```

Comment:
Ideally this would just be `np.broadcast_to(color, filled.shape + color.shape)`, but that's introduced in numpy 1.10